### ISSUE TITLE
Support for Steam Controllers

### ISSUE THREAD

**Issue by johnjelinek** [xboxdrv/xboxdrv#202]
Is there any documentation on mapping multiple steam controllers connected to a single wireless dongle?

**Comment by chewi** [comment]
I've just acquired a Stream controller so I'm going to attempt to add native support. It is obviously important to consider this feature when adding such support but I only have one controller. I gather that the Xbox 360 dongle can also do this though and I'm not aware of any support for this in the code. I'll see what that looks like with two controllers.

**Comment by Grumbel** [comment]
There is a Open Source driver for the Steam Controller at:

https://github.com/ynsta/steamcontroller

**Comment by chewi** [comment]
Yeah, I do know about it and saw last night that it uses different USB interface/endpoints for multiple controllers, which makes sense. I don't know if you were implying that support isn't needed here but I think supporting it through xboxdrv would be much better!

**Comment by chewi** [comment]
Just to let you know that I already have the single controller mostly working! :tada: Will polish it off later. Oh and for some reason, Xorg suddenly detects it as a mouse when xboxdrv starts. I'm guessing this is because `--detach-kernel-driver` detaches the generic HID endpoint but not the mouse endpoint and so Xorg figures it must just be a regular mouse.

**Comment by johnjelinek** [comment]
Great news!

**Comment by chewi** [comment]
I've pushed up my work so far to my fork under the steam-controller branch.

All the controls are mapped out. The d-pad buttons don't allow you to press up/down together with left/right so I emulated it by checking the analog pad position when the buttons are pressed. 

It should work with wired and multiple wireless controllers in the same manner that multiple 360 controllers are currently supported.

I've been having mixed fortunes with force feedback. I tried to copy what the Python project does and I did get it to work just once but it seems very temperamental. Sometimes the controller won't turn back on until I pop a battery out.

**Comment by chewi** [comment]
See #222.

**Comment by Grumbel** [comment]
Steam Controller support has been merged, but needs testing or getting hidden being an experimental flag.
